source/administration/plugins.rst

```diff
@@ -12,12 +12,63 @@ Consider using a plugin in the following scenarios:
+
+Configuring and enabling a plugin
+~~~~~~~~~~~~~~~
+
+Once a plugin has been installed (or pre-installed if it shipped with Mattermost binary release) there will be a "Configure" button shown next to the plugin listing.  
+
+1. Click on the button, you will be navigated to the plugins "Settings" page.  
+2. Fill in Plugin settings as required
+3. At the top of the page, set **Enable Plugin** setting to ``True``.  If this flag is not switched on, the plugin will never become active.
+4. Test out the plugin as needed
+
+Upgrading Plugins
+~~~~~~~~~~~~~~~~~
+
+In v5.16, the Marketplace only supports the installation of new plugins.  To upgrade a plugin, you will need to manually update it by downloading the binary file from the GitHub repository and uploading it in the **System Console > Plugin management** section.  In v5.18 (Dec 2019), the Marketplace will allow you to upgrade a plugin on demand when a new version becomes available.
+
+Marketplace Server
+-----
+
+There are two Marketplace settings in the **System Console > Plugin management** section:
+
+.. image:: https://user-images.githubusercontent.com/915956/66892854-94660d80-efa1-11e9-805c-a85223d43a07.png
+
+- **Enable Marketplace** Turns the Marketplace UI on or off for sysadmins
+- **Marketplace URL** The location of the Marketplace Server you want to query for new plugins. Mattermost hosts a Plugin Marketplace for the community and is the default value for this field. You can setup your own Marketplace server if you want to rely on your own infrastructure.
+
+When you are first presented with the Marketplace, your Mattermost server will attempt to contect the Marketplace Server, run by Mattermost, and will be returned a list of available plugins that are appropriate based on the server version that is currently running.  Only your server version and search query is passed over to Mattermost Marketplace, we retain an anonymized record for product analytics whenever a new plugin is installed, unless you have opted out of `Telemetry <https://docs.mattermost.com/administration/telemetry.html>`__ previously. The `Plugin marketplace server code <https://github.com/mattermost/mattermost-marketplace>`__ is available as an open-source project and can be used to setup your own private Marketplace if desired.  
+
+Which plugins are listed in the Marketplace?
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The `Mattermost Plugin Marketplace <https://github.com/mattermost/mattermost-marketplace>`__ is a service run by Mattermost that contains listings of plugins that we have reviewed and in many cases built. In the future, we plan to include community developed plugins that will be labeled differently than "mattermost developed" plugins.  We plan to include settings that would restrict which types of plugins you can install.  Comments in our forum are welcome as we develop this feature further. 
```

Comment:
Add a link to forum: https://forum.mattermost.org/.

Reply:
done